classes/middleware/PKPAuthenticateSession.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @file classes/middleware/PKPAuthenticateSession.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPAuthenticateSession
+ *
+ * @ingroup middleware
+ *
+ * @brief 
+ */
+
+namespace PKP\middleware;
+
+use PKP\security\Validation;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Http\Request;
+
+class PKPAuthenticateSession extends \Illuminate\Session\Middleware\AuthenticateSession
+{
+    /**
+     * Log the user out of the application.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return void
+     */
+    protected function logout($request)
+    {
+        Auth::logout();
+
+        $request->session()->invalidate();
+        $request->session()->regenerateToken();
+
+        return $this->redirectTo($request);
+    }
+
+    /**
+     * Get the path the user should be redirected to when their session is not authenticated.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return string|null
+     */
+    protected function redirectTo(Request $request)
+    {
+        Validation::redirectLogin();
```

Comment:
One thing we can do now that we're relying more on Laravel's tools is decrease the use of the `Validation` class, which has been a bit of a dumping ground for functions over the years. If we can deprecate `Validation` and have code that would normally call it rely directly on the Laravel tools, I'd prefer that.

Reply:
To apply laravel toolset, more specifically the `redirection` service which is dependent on routing system which not yet implemented for the Page and Component routers . I also think the `Validation` class should be removed at one point but that should be the part of bigger task like integrating the routing system which will allow us to move a log these code to routing middleware .